File: middleware/src/dsl/GoalDependencies.java
package dsl;

import java.util.ArrayList;
import java.util.List;

public class GoalDependencies {
	private List<Goal> dependencies = new ArrayList<Goal>(); 
	double earliestStartTime;
	double latestFinishTime;
	
	public GoalDependencies(double earliestStartTime, double latestFinishTime) {
		this.earliestStartTime = earliestStartTime;
		this.latestFinishTime = latestFinishTime;
	}
	
	public void addDependency(Goal dependency) {
		dependencies.add(dependency);
	}
	
	public boolean isReady(double timeNow) {
		boolean depsReady = true;
		
		for (Goal d : dependencies) {
			if (!(d.isReady(timeNow)))
				depsReady = false;
		}
		return depsReady && (timeNow >= earliestStartTime) && (timeNow <= latestFinishTime); 
	}

	public boolean isLate(double timeNow) {
		return (timeNow > latestFinishTime);
	}

	public List<Goal> getDependencies() {
		return dependencies;
	}
}

File: middleware/dsl/mission/GoalDependencies.py
class GoalDependencies:
	
	def __init__(self, depts):
		self.dependencies = []
		self.dependencies = depts

	def add_dependency(self, dependency):
		self.dependencies.add(dependency)

	def is_ready(self, timeNow):
		depsReady = True	
		for goal in self.dependencies:
			if not goal.isReady(timeNow):
				depsReady = False
		return depsReady and (timeNow >= self.earliestStartTime) and (timeNow <= self.latestFinishTime)

	def is_late(self, timeNow):
		return timeNow > self.latestFinishTime

	def get_dependencies(self):
		return self.dependencies